src/main/java/org/stellar/anchor/paymentservice/circle/CirclePaymentsService.java

```diff
@@ -257,6 +260,73 @@ private HttpClient getWebClient(boolean authenticated) {
+        if (sourceAccount.network != Network.CIRCLE) {
+            throw new HttpException(400, "the only supported network for the source account is circle");
+        }
+        if (!List.of(Network.CIRCLE, Network.STELLAR).contains(destinationAccount.network)) {
+            throw new HttpException(400, "the only supported networks for the destination account are circle and stellar");
+        }
+        if (!currencyName.startsWith(destinationAccount.network.getCurrencyPrefix())) {
+            throw new HttpException(400, "the currency to be sent must contain the destination network schema");
+        }
+    }
+
+    /**
+     * API request that sends a Circle transfer, i.e. a payment from a Circle account to another Circle account or to a
+     * blockchain wallet.
+     *
+     * @param sourceAccount      the account where the payment will be sent from.
+     * @param destinationAccount the account that will receive the payment, it can be an internal Circle account or a Stellar wallet.
+     * @param balance            the balance to be transferred.
+     * @return asynchronous stream with the payment object.
+     * @throws HttpException If the Circle http response status code is 4xx or 5xx.
+     * @throws HttpException If the destination network is not a Circle account nor a Stellar wallet.
+     */
+    private Mono<Payment> sendTransfer(Account sourceAccount, Account destinationAccount, CircleBalance balance) throws HttpException {
+        CircleTransactionParty source = CircleTransactionParty.wallet(sourceAccount.getId());
+        CircleTransactionParty destination;
+        switch (destinationAccount.network) {
+            case CIRCLE:
+                destination = CircleTransactionParty.wallet(destinationAccount.getId());
+                break;
+            case STELLAR:
+                destination = CircleTransactionParty.stellar(destinationAccount.id, destinationAccount.getIdTag());
+                break;
+            default:
+                throw new HttpException(400, "the destination network is not supported for Circle transfers");
+        }
+        CircleSendTransactionRequest req = CircleSendTransactionRequest.forTransfer(source, destination, balance);
+        String jsonBody = gson.toJson(req);
+
+        return getWebClient(true)
+                .post()
+                .uri("/v1/transfers")
+                .send(ByteBufMono.fromString(Mono.just(jsonBody)))
+                .responseSingle((postResponse, bodyBytesMono) -> {
+                    if (postResponse.status().code() >= 400) {
+                        return handleCircleError(postResponse, bodyBytesMono);
+                    }
+
+                    return bodyBytesMono.asString();
+                }).flatMap(body -> {
+                    CircleTransferResponse transfer = gson.fromJson(body, CircleTransferResponse.class);
+                    return Mono.just(transfer.getData().toPayment());
+                });
```

Comment:
Are we certain that all responses with a status code below 400 contains the expected json body? We may want to have some logic that handles when an unexpected response is returned.

Reply:
You're right, we're not handling the case when the response status code is "fine" but the response body is totally unexpected.

Creating an issue for that: https://github.com/stellar/java-stellar-anchor-sdk/issues/14